e2e_playwright/st_download_button_deferred_test.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2025)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+
+def test_deferred_download_csv_success(app: Page):
+    """Test successful deferred CSV download."""
+    # Get the CSV download button (second button in the app)
+    csv_button = app.get_by_role("button", name="Download CSV (Deferred)")
+    expect(csv_button).to_contain_text("Download CSV (Deferred)")
+
+    # Click and expect download
+    with app.expect_download() as download_info:
+        csv_button.click()
+
+    download = download_info.value
+    assert download.suggested_filename == "data.csv"
+
+    # Verify downloaded content
+    content = download.path().read_text()
+    assert "Name,Age,City" in content
+    assert "Alice,30,NYC" in content
+    assert "Bob,25,LA" in content
+
+
+def test_deferred_download_lambda_text(app: Page):
+    """Test deferred download with lambda function."""
+    # Get the lambda download button (third button)
+    lambda_button = app.get_by_role("button", name="Download Lambda Text")
+    expect(lambda_button).to_contain_text("Download Lambda Text")
```

Comment:
nit: I think checking the text here again is a bit redundant since we are already accessing via the text. Just a `expect(lambda_button).to_be_visible()` should be fine and also checks for actual screen visibility.